### PR TITLE
203: Support NO_COLOR

### LINKED ISSUE #203: Support NO_COLOR
An increasing number of command-line software programs output text with [ANSI color](https://en.wikipedia.org/wiki/ANSI_escape_code#Colors) escape codes by default. While some developers and users obviously prefer seeing these colors, some users don’t. Unfortunately, every new piece of software seems to have a [different way](https://no-color.org/#other) of disabling colored text output and some software has no way at all.

Accepting the futility of trying to reverse this trend, an informal standard was proposed in 2017:

Command-line software which adds ANSI color to its output by default should check for a NO_COLOR environment variable that, when present and not an empty string (regardless of its value), prevents the addition of ANSI color.

By adopting this standard, users that prefer to have plain, non-colored text output can export NO_COLOR=1 to their shell’s environment and automatically disable color by default in all supported software.

If your software outputs color by default, please consider not doing so. If you insist, please implement this standard to make it easy for your users to disable color and then add your software to this list by [submitting a pull request](https://github.com/jcs/no_color).

See: https://no-color.org/

### PR DESCRIPTION
**Pull request by DanielJette**
### What does this change accomplish?

Resolves #203 

### How have you achieved it?

Add support for `NO_COLOR`. See https://no-color.org/ for more information.

### Scope of Impact and Testing instructions

Set `export NO_COLOR=true` and run various testify command (ex. `./gradlew LegacySample:testifySettings`) and verify that no color is used in the output.

### Notice

> [!WARNING]
> This change must keep `main` in a shippable state; **it may be shipped without further notice**.

### CODE REVIEW

**Review by lwise — APPROVED** [review]

File: Plugins/Gradle/src/main/kotlin/dev/testify/tasks/utility/SettingsTask.kt [hunk 1]
```diff
@@ -96,8 +134,11 @@ open class SettingsTask : TestifyUtilityTask() {
         println("  testRunner             = $testRunner")
         println("  useSdCard              = $useSdCard")
         println("  useTestStorage         = $useTestStorage")
-        println("  isRecordMode           = $isRecordMode")
         println("  user                   = $userId")
+
+        val notSet = "${AnsiFormat.Bold}${AnsiFormat.Green}Not set${AnsiFormat.Reset}"
```
Comment (jaredh) [inline Plugins/Gradle/src/main/kotlin/dev/testify/tasks/utility/SettingsTask.kt:139]: Nice touch.

**Review by jaredh — APPROVED** [review]
TIL NO_COLOR.

✅ Tested

Might be useful to respect Gradle's console output mode for this as well.

**Comment by DanielJette** [comment]
> Might be useful to respect Gradle's console output mode for this as well.

@jaredh I haven't heard of Gradle's console output mode before. Do you mean this? https://docs.gradle.org/current/userguide/command_line_interface.html#sec:command_line_customizing_log_format

**Comment by jaredh** [comment]
> > Might be useful to respect Gradle's console output mode for this as well.
> 
> @jaredh I haven't heard of Gradle's console output mode before. Do you mean this? [docs.gradle.org/current/userguide/command_line_interface.html#sec:command_line_customizing_log_format](https://docs.gradle.org/current/userguide/command_line_interface.html#sec:command_line_customizing_log_format)

@DanielJette Yep, thats the one. `plain` should have no colour.

**Comment by DanielJette** [comment]
https://github.com/ndtp/android-testify/pull/239 is a better solution

**Pull request closed without merging**